ducktape/services/service.py

```diff
@@ -12,36 +12,63 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Service classes know how to deploy a service onto a set of nodes and then
-# clean up the services. They request the necessary resources from the cluster,
-# configure each server, and bring up/tear down the service. They also expose
-# information about the service so that other services or test scripts can
-# easily be configured to work with them. Finally, they may be able to collect
-# and check logs/output from the service, which can be helpful in writing tests
-# or benchmarks.
-#
-# Services should generally be written to support an arbitrary number of nodes,
-# even if instances are independent of each other. They should be able to assume
-# that there won't be resource conflicts: the cluster tests are being run on
-# should be large enough to use one instance per service instance.
 
 from ducktape.command_line.config import ConsoleConfig
 
 
 class Service(object):
-    def __init__(self, service_context):
-        """
-        :type service_context: ServiceContext
-        """
-        self.num_nodes = service_context.num_nodes
-        self.cluster = service_context.cluster
-        self.logger = service_context.logger
+    """Service classes know how to deploy a service onto a set of nodes and then clean up after themselves.
 
-    def start(self):
-        """Start the service running in the background."""
+    They request the necessary resources from the cluster,
+    configure each node, and bring up/tear down the service.
+
+    They also expose
+    information about the service so that other services or test scripts can
+    easily be configured to work with them. Finally, they may be able to collect
+    and check logs/output from the service, which can be helpful in writing tests
+    or benchmarks.
+
+    Services should generally be written to support an arbitrary number of nodes,
+    even if instances are independent of each other. They should be able to assume
+    that there won't be resource conflicts: the cluster tests are being run on
+    should be large enough to use one instance per service instance.
+    """
+
+    def __init__(self, context, num_nodes):
```

Comment:
Might want to add a small docstring that at least explains the `context` argument. I think the explanation just below here is fine, and maybe just add that in most cases, the context will be your test class instance (i.e. self in your Test).

We haven't really talked about how to handle docs for ducktape yet, but writing a new Service is probably the first thing any user needs to do, so they'll hit this pretty quickly.